core/master/src/main/java/org/drftpd/commands/usermanagement/GroupManagementHandler.java

```diff
@@ -0,0 +1,864 @@
+	 * only show up in "site flags" if they're turned on.
+	 * <p>
+	 * ex. site change Archimede ratio 5
+	 * <p>
+	 * This would set the ratio to 1:5 for the user 'Archimede'.
+	 * <p>
+	 * ex. site change Archimede flags +2-AG
+	 * <p>
+	 * This would make the user 'Archimede' groupadmin and remove his ability to
+	 * use the commands site nuke and site give.
+	 * <p>
+	 * NOTE: The flag DELETED can not be changed with site change, it will
+	 * change when someone does a site deluser/readd.
+	 *
+	 * @throws ImproperUsageException
+	 */
+	public CommandResponse doSITE_CHANGEGROUP(CommandRequest request) throws ImproperUsageException {
+
+		if (!request.hasArgument()) {
+			throw new ImproperUsageException();
+		}
+
+		Collection<Group> groups = new ArrayList<>();
+
+		Group groupToChange;
+		CommandResponse response = StandardCommandManager.genericResponse("RESPONSE_200_COMMAND_OK");
+		Map<String, Object> env = new HashMap<>();
+
+		StringTokenizer arguments = new StringTokenizer(request.getArgument());
+
+		if (!arguments.hasMoreTokens()) {
+			throw new ImproperUsageException();
+		}
+
+		String groupname = arguments.nextToken();
+
+		try {
+			if (groupname.equals("*")) {
+				// This request is to change all users
+				groups = GlobalContext.getGlobalContext().getUserManager().getAllGroups();
+			} else {
+				// Get the actual group
+				groups.add(GlobalContext.getGlobalContext().getUserManager().getGroupByNameUnchecked(groupname));
+			}
+		} catch (NoSuchGroupException e) {
+			return new CommandResponse(550, "Group " + groupname + " not found: " + e.getMessage());
+		} catch (GroupFileException e) {
+			logger.log(Level.ERROR, "Error loading group", e);
+
+			return new CommandResponse(550, "Error loading group: " + e.getMessage());
+		}
+
+		if (!arguments.hasMoreTokens()) {
+			throw new ImproperUsageException();
+		}
+
+		String command = arguments.nextToken().toLowerCase();
+
+		Session session = request.getSession();
+
+		User currentUser = session.getUserNull(request.getUser());
+
+		boolean isAdmin = currentUser.isAdmin();
+
+		// Only Site Admins are allowed here
+		if (!isAdmin) {
+			return StandardCommandManager.genericResponse("RESPONSE_530_ACCESS_DENIED");
+		}
+
+		// String args[] = request.getArgument().split(" ");
+		// String command = args[1].toLowerCase();
+		// 0 = user
+		// 1 = command
+		// 2- = argument
+		String[] commandArguments = new String[arguments.countTokens()];
+		String fullCommandArgument = "";
+
+		for (int x = 0; arguments.hasMoreTokens(); x++) {
+			commandArguments[x] = arguments.nextToken();
+			fullCommandArgument = fullCommandArgument + " " + commandArguments[x];
+		}
+
+		fullCommandArgument = fullCommandArgument.trim();
+
+		for (Group group1 : groups) {
+			groupToChange = group1;
+			env.put("targetgroup", groupToChange.getName());
+
+			switch (command) {
+				//TODO: Figure out what this is used for?
+/*
+                case "ratio":
```

Comment:
Please no dead code.

Reply:
Fixed as I finally understand what group ratio is used for.